web/src/lib/components/elements/buttons/circle-icon-button.svelte

```diff
@@ -1,19 +1,32 @@
+<script lang="ts" context="module">
+  export type Color = 'transparent' | 'light' | 'dark' | 'gray' | 'primary' | 'opaque';
+</script>
+
 <script lang="ts">
   import Icon from '$lib/components/elements/icon.svelte';
-  type Color = 'transparent' | 'light' | 'dark' | 'gray' | 'primary' | 'opaque';
 
   export let type: 'button' | 'submit' | 'reset' = 'button';
   export let icon: string;
   export let color: Color = 'transparent';
   export let title: string;
+  /**
+   * The padding of the button, used by the `p-{padding}` Tailwind CSS class.
+   */
   export let padding = '3';
+  /**
+   * Size of the button, used for a CSS value.
+   */
   export let size = '24';
   export let hideMobile = false;
   export let buttonSize: string | undefined = undefined;
   /**
    * viewBox attribute for the SVG icon.
    */
   export let viewBox: string | undefined = undefined;
+  export let id: string | undefined = undefined;
+  export let ariaHasPopup: boolean | undefined = undefined;
+  export let ariaExpanded: boolean | undefined = undefined;
+  export let ariaControls: string | undefined = undefined;
```

Comment:
I think there is an export HTML*Type field you can use to get the actual type for these properties.

Reply:
In my opinion, I find the syntax for automatically adding the HTML props a little confusing/cumbersome, especially because this `CircleIconButton` component has so many props. Here's a similar example:

https://github.com/immich-app/immich/blob/08426e14bcdb3bd8bbad5d778df7b78cd4da5857/web/src/lib/components/shared-components/immich-logo.svelte#L13-L19

I'm wondering if this would be an easier change to do once we have the new Svelte 5 prop typing? Happy to add the automatic HTML props now if that's preferred.

https://svelte-5-preview.vercel.app/docs/runes#$props